examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -131,6 +164,13 @@ int main(int argc, char **argv)
   //std::cout << "Calling cudaFree... " << std::endl;
   checkCuda( cudaFree( 0 ) ); // SLOW!
   //std::cout << "Calling cudaFree... done" << std::endl;
+
+  // --- Book the tear down at the end of main:
+  struct CudaTearDown {
+    ~CudaTearDown() {
+      checkCuda( cudaDeviceReset() ); // this is needed by cuda-memcheck --leak-check full
```

Comment:
Maybe add an std::cout commented out about calling this cudadevicereset, may be useful for later debugging

Reply:
Should I add this now? That's maybe good if `-d` is active, but it shouldn't print in general, should it?

Reply:
I did add it, but only in `-d` mode.

Reply:
thanks  - you can even ompit the _print and just comment it out, both are ok